lib/scala/profiling-utils/src/main/java/org/enso/profiling/sampler/OutputStreamSampler.java

```diff
@@ -0,0 +1,56 @@
+package org.enso.profiling.sampler;
+
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import org.netbeans.modules.sampler.Sampler;
+
+/**
+ * Gathers application performance statistics that can be visualised in Java VisualVM, and writes it
+ * to the provided output.
+ */
+public final class OutputStreamSampler implements MethodsSampler {
+
+  private final Sampler sampler = Sampler.createSampler(this.getClass().getSimpleName());
+  private final OutputStream outputStream;
+
+  private boolean isSamplingStarted = false;
+
+  /**
+   * Creates the {@link OutputStreamSampler} for provided output stream.
+   *
+   * @param outputStream the output stream to write result to.
+   */
+  public OutputStreamSampler(OutputStream outputStream) {
+    this.outputStream = outputStream;
+  }
+
+  public static OutputStreamSampler ofFile(File file) throws FileNotFoundException {
+    return new OutputStreamSampler(new FileOutputStream(file));
```

Comment:
This means the stream is open during the whole sampling rather than being opened and close just for the time of `stop`ing and writing the data down.